control/init.sh

```diff
@@ -17,10 +17,20 @@ function realpath() {
 
 
 if [ -z ${CI_TEST} ]; then
-    if [ ! -f $VENV/bin/activate ]; then
-        if [[ $CCHQ_VIRTUALENV == *3.10* ]]; then
-          # use venv because 3.10 setup includes installing python3.10-venv
-          python3.10 -m venv $VENV
+    if [[ ! -f $VENV/bin/activate ]]; then
+        if [[ -f /usr/bin/python3.10 ]]; then
```

Comment:
I assume the actual installed location of the `python3.10` binary isn't important? If it must be exactly `/usr/bin/python3.10` then it would be good to document why that is. Otherwise, using `hash` to check for binaries on PATH is clearer that _"any Python 3.10 binary will do"_.

```suggestion
        if hash python3.10 2>/dev/null; then
```

Reply:
Yeah I wasn't thinking of cases where it would be installed elsewhere, but agreed that isn't important so this is a better check. Thanks!

Reply:
> I wasn't thinking of cases where it would be installed elsewhere

:+1: Yep, this is why I asked. Some further elaborations if you care to understand the perspective I was coming from:

If this script is worried about the presence of "unexpected" python3.10 installations, then it (this script) would need to both:

1. Check for the existence of the version it cares about (presumably the source line of this thread), and
2. explicitly use the expected version everywhere it calls `python3.10 ...`.

Alternatively, it (what you've done here) is a much cleaner "separation of interests" for this script to "not care" which python3.10 it uses, and just assume that another "machine setup scope" has taken care to ensure the installed version of python3.10 is "safe to use without further consideration".